DSharpPlus/Entities/Emoji/DiscordEmoji.cs

```diff
@@ -141,7 +141,7 @@ public bool Equals(DiscordEmoji e)
             if (e is null)
                 return false;
 
-            return ReferenceEquals(this, e) ? true : this.Id == e.Id && this.Name == e.Name;
+            return ReferenceEquals(this, e) || (this.Id == e.Id && (this.Id != 0 || this.Name == e.Name));
```

Comment:
Couldn't this be simplified to `|| (this.Id == e.Id || this.Name == e.Name)`?

Reply:
Wait nevermind ignore me

Reply:
what about 
```csharp
        public bool Equals(DiscordEmoji e)
        {
            return e is null ? false : ReferenceEquals(this, e) ? true : this.Id == e.Id;
        }
```

and

```csharp
        public static bool operator ==(DiscordEmoji e1, DiscordEmoji e2)
        {
            var o1 = e1 as object;
            var o2 = e2 as object;

            return (o1 == null && o2 != null) || (o1 != null && o2 == null)
                ? false
                : o1 == null && o2 == null ? true : e1.Id == e2.Id;
        }
        ```

Reply:
The top one is bad because any unicode emoji is equal to any other unicode emoji, since they all have the id of `0`

Reply:
> ```cs
>            return (o1 == null && o2 != null) || (o1 != null && o2 == null)
>               ? false
>                : o1 == null && o2 == null ? true : e1.Id == e2.Id;
>```

This is a clusterfuck

Reply:
> The top one is bad because any unicode emoji is equal to any other unicode emoji, since they all have the id of `0`

Ah okay. Didn't thought about this.